### ISSUE TITLE
struggling to override basic templates

### ISSUE THREAD

**Issue by beatelite** [marmelab/ng-admin#914]
First, thanks for all the amazing work. I have been building my own app but switched to your admin app due it's power in creating entities and related views very easily. 

I found setting up the entities and related fields to be a snap (thanks to your guide). However, customizing the creation pages is really a challenge for me. 

on my post entity I set up a creation view with a custom template like so:

```
    var articleTemplate = require('text!../tpl/articleView.html');
   post.creationView()
   .template(articleTemplate)
  .fields([
        nga.field('title'),
        nga.field('content'),
        nga.field('status', 'choice')
          .choices([
            { value: 'publish', label: 'Publish' },
            { value: 'draft', label: 'Draft' },
            { value: 'pending', label: 'Pending Review' },
            { value: 'future', label: 'Publish In The Future' },
          ])

    ]);
```

The template looks like this:

```
<div class="row">
    <div class="col-lg-12">
        <div class="page-header">
            <ma-view-actions override="::formController.actions" entry="entry" entity="::formController.entity">
                <ma-list-button ng-if="::entity.listView().enabled" entity="::entity"></ma-list-button>
                <ma-delete-button ng-if="::entity.deletionView().enabled" entry="entry" entity="::entity"></ma-delete-button>
            </ma-view-actions>
            <p class="lead" ng-if="::formController.description" compile="::formController.description">{{ ::formController.description }}</p>
        </div>
    </div>
</div>

<div class="row ng-admin-entity-post" id="edit-view">
    <form class="col-lg-12 form-horizontal" name="formController.form" ng-submit="formController.submitEdition($event)">
        <div class="row">
            <div class="col-lg-6">


                <div ng-init="field=formController.fields[0]" id="row-reference" class="form-group has-feedback" ng-class="formController.form.title.$dirty ? (formController.form.title.$valid ? 'has-success' : 'has-error') : ''">
                    <label for="title" class="col-sm-4 control-label">Title&nbsp;*&nbsp;</label>
                    <div class="col-sm-8">
                        <ma-input-field field="::field" value="post.title.rendered"></ma-input-field>
                    </div>
                </div>
            </div>   
        </div>
        <div class="form-group">
            <div class="col-sm-offset-4 col-sm-8">
                <button type="submit" class="btn btn-primary"><span class="glyphicon glyphicon-ok"></span> Save Changes</button>
            </div>
        </div>
    </form>
</div>
```

I can see the single field and the submit button. When I enter data and submit, I get a "404 rest no route" error. When I look I see it's sending a PUT request (even though I'm on /create state.)

If I remove the template() function in the entity config, the regular fields show up and the form sends a POST request as it should. 

This is really got my head ringing. What am I doing wrong?

Also, do you have any step by step guides for theming like you do for the entity creation and manipulation? Those were a tremendous help. I feel like I'm fumbling through the theming portion by looking at your demo apps and while those are complete, they don't really explain how or why for certain things.

**Comment by beatelite** [comment]
I did. That's actually where I borrowed the code in my example from. Any idea why it's sending a put request?

**Comment by beatelite** [comment]
Ok, I get most of how this works. I just needed to look at the example code a bit more clearly. I noticed that it references

ng-init="field=formController.fields[0]"

which is the index of the field based on where it is in the config file for that view.

I can get everything lined up and even get a response back saying I'm using the right fields on the API. However, everytime I try a custom form, it sends a PUT request when it should be a POST (since I'm on the /create view). 

Any idea or direction here?

Thanks for all your help. I'm slowly slogging my way through understanding everything. It's a great piece of work. ;)
